crates/guest-rust/rt/src/async_support/cabi.rs

```diff
@@ -0,0 +1,105 @@
+//! Definition of the "C ABI" of how imported functions interact with exported
+//! tasks.
+//!
+//! Ok this crate is written in Rust, why in the world does this exist? This
+//! comment is intended to explain this rationale but the tl;dr; is we want
+//! this to work:
+//!
+//! * Within a single component ...
+//! * One rust crate uses `wit-bindgen 0.A.0` to generate an exported function.
+//! * One rust crate uses `wit-bindgen 0.B.0` to bind an imported function.
+//! * The two crates are connected in the application with
+//!   `std::future::Future`.
+//!
+//! Without this module this situation won't work because 0.A.0 has no
+//! knowledge of 0.B.0 meaning that when 0.B.0 decides to block it won't know
+//! where to register its `waitable` within a `waitable-set`.
+//!
+//! To solve this problem the long-term intention is that something will live
+//! in `wasi-libc` itself, but in the meantime it's living "somewhere" within
+//! `wit-bindgen 0.*.0`. Specifically all `wit-bindgen` versions will all
```

Comment:
"all ... versions will all" seems redundant. Sorry for the nit, it's all I'm good at these days in your fast moving project.